curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderLatch.java

```diff
@@ -149,6 +141,7 @@ public LeaderLatch(CuratorFramework client, String latchPath, String id, CloseMo
         this.latchPath = PathUtils.validatePath(latchPath);
         this.id = Preconditions.checkNotNull(id, "id cannot be null");
         this.closeMode = Preconditions.checkNotNull(closeMode, "closeMode cannot be null");
+        listener = client.decorateConnectionStateListener((__, newState) -> handleStateChange(newState));
```

Comment:
Note: I'm going to make the decoration automatic. It's silly to force this on clients.